test/integration/crud/crud.prose.test.js

```diff
@@ -12,32 +13,127 @@ describe('CRUD Prose Spec Tests', () => {
+    /**
+     * Test that writeConcernError.errInfo in a command response is propagated as WriteConcernError.details (or equivalent) in the driver.
+     * Using a 4.0+ server, set the following failpoint:
+     * {
+     *   "configureFailPoint": "failCommand",
+     *   "data": {
+     *     "failCommands": ["insert"],
+     *     "writeConcernError": {
+     *       "code": 100,
+     *       "codeName": "UnsatisfiableWriteConcern",
+     *       "errmsg": "Not enough data-bearing nodes",
+     *       "errInfo": {
+     *         "writeConcern": {
+     *           "w": 2,
+     *           "wtimeout": 0,
+     *           "provenance": "clientSupplied"
+     *         }
+     *       }
+     *     }
+     *   },
+     *   "mode": { "times": 1 }
+     * }
+     *
+     * Then, perform an insert operation and assert that a WriteConcernError occurs and that
+     * its details property is both accessible and matches the errInfo object from the failpoint.
+     */
+    metadata: { requires: { mongodb: '>=4.0.0' } },
     async test() {
+      throw new Error('This test is not implemented!');
+    }
+  }).skipReason = 'TODO(NODE-3888): Implement this test';
+
+  describe('2. WriteError.details exposes writeErrors[].errInfo', () => {
+    /**
+     * Test that writeErrors[].errInfo in a command response is propagated as WriteError.details (or equivalent) in the driver.
+     * Using a 5.0+ server, create a collection with document validation like so:
+     * {
+     *   "create": "test",
+     *   "validator": {
+     *     "x": { $type: "string" }
+     *   }
+     * }
+     *
+     * Enable command monitoring to observe CommandSucceededEvents.
+     * Then, insert an invalid document (e.g. {x: 1})
+     * and assert that a WriteError occurs, that its code is 121 (i.e. DocumentValidationFailure),
+     * and that its details property is accessible.
+     * Additionally, assert that a CommandSucceededEvent was observed
+     * and that the writeErrors[0].errInfo field in the response document matches the WriteError's details property.
+     */
+
+    let collection;
+    beforeEach(async () => {
       try {
         await client.db().collection('wc_details').drop();
       } catch {
         // don't care
       }
 
-      const collection = await client
+      collection = await client
         .db()
         .createCollection('wc_details', { validator: { x: { $type: 'string' } } });
+    });
 
-      try {
-        await collection.insertMany([{ x: /not a string/ }]);
-        expect.fail('The insert should fail the validation that x must be a string');
-      } catch (error) {
-        expect(error).to.be.instanceOf(MongoBulkWriteError);
-        expect(error).to.have.property('code', 121);
-        expect(error).to.have.property('writeErrors').that.is.an('array');
-        expect(error.writeErrors[0]).to.have.property('errInfo').that.is.an('object');
+    it('test case: insert MongoServerError', {
```

Comment:
I think this is "1. WriteConcernError.details exposes writeConcernError.errInfo" and same with the test below, unless I'm missing something they're supposed to be asserting